master/buildbot/db/connector.py

```diff
@@ -162,15 +162,17 @@ def set_master(self, master):
 
     @defer.inlineCallbacks
     def setup(self, check_version=True, verbose=True):
-        if self.configured_url is None:
-            self.configured_url = yield self.master.get_db_url(self.master.config)
+        if self.configured_db is None:
+            self.configured_db = yield self.master.get_db_config(self.master.config)
 
-        db_url = self.configured_url
+        db_config = dict(self.configured_db)
```

Comment:
I think that we should put sqlalchemy kwargs into separate key. E.g. `engine_kwargs` to make it clear where they are used. This would make buildbot config structure more self descriptive.